string[] objects
Pose2D[] coordinates